dot/core/service.go

```diff
@@ -200,10 +207,12 @@ func NewService(cfg *Config) (*Service, error) {
 			storageState:     cfg.StorageState,
 			transactionQueue: cfg.TransactionQueue,
 			isBabeAuthority:  false,
+			lock:             chanLock,
 			closed:           false,
 			syncer:           syncer,
 			syncLock:         syncerLock,
 			syncChan:         cfg.SyncChan,
+			respChan:         respChan,
```

Comment:
does this need to be appended to both service and and syncer submodule?

Reply:
see comment below